proposed/2022/Package-Source-Mapping-VS-Options.md

```diff
@@ -0,0 +1,133 @@
+
+**Example** 
+
+```xml
+<PackageReference Include="Microsoft.A" Version="1.0.0"/>
+<PackageReference Include="Newtonsoft.Json" Version="11.0.0"/>
+<PackageReference Include="Serilog" Version="11.0.0"/>
+```
+
+```xml
+<packageSourceMapping>
+    <packagesource key="nuget.org">
+        <package pattern="Serilog" />
+        <package pattern="Microsoft.Extensions.Options" />
+        <package pattern="Newtonsoft.Json" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+**Result:**
+
+In this example, the user added `Serilog`, `Microsoft.A`, and `Newtonsoft.Json`, separately, on the package source mappings page and selected `nuget.org` as the source from the `ComboBox` for all of them. All three package source mappings will be configured to be installed from `nuget.org`. 
+
+If the user removes one of these package source mappings, it will be deleted from the configuration as well. For example, if the user removes `Serilog`, the above code becomes
+
+```xml
+<packageSourceMapping>
+    <packagesource key="nuget.org">
+        <package pattern="Microsoft.Extensions.Options" />
+        <package pattern="Newtonsoft.Json" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+If the user clicks the _Clear_ button on the package source mappings options page, then all package sources will be removed from the configuration, and package source mapping will be disabled. 
+
+If the user removes a source from the package sources page, any mappings to that source will not be deleted in the config. The user will get a message that they are removing a source that still has mappings to it. If a user adds a source that has already has mappings to it in the config, this will also be a message to the user.
+
+#### Package Sources Page
+
+When the user opens the package sources page, it will automatically display a pin icon with all previously pinned default sources.
+
+When the user pins a package source(s), the NuGet config will be updated. Unless otherwise specified in the package source mappings page, all package IDs will be pinned to this source because the `pattern=*` in the config. 
+
+**Example**
+
+```xml
+<packageSourceMapping>
+    <packagesource key="contoso">
+        <package pattern="*" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+**Result:**
+
+In this example, the user pinned the source `contoso` and did not specify any specific package source mappings. All packages will come from the source `contoso` due to matching the `*` pattern.
+
+### Accessibility 
+
+To ensure screen-reader I will run Accessibility Insights to find any accessibility faults.
+
+
+## Drawbacks 
+
+Currently, mappings to sources are not deleted in the config file when the source they are mapped to is deleted. It is complicated to remove this code because there are multiple different config files this mapping could be in.

```

Comment:
Would it be possible to show in any of the options page the filepath of the modified nuget.config file?